### PR TITLE
feat: Add Guidelines for SDK Setup Wizards

### PR DESCRIPTION
**Pull request by Lms24**
This PR adds an SDK Setup Wizards page to the SDK development documentation.

The purpose of this page is to 
1. Formulate high-level requirements for setup wizards
2. Provide instructions around implementing new wizards such as the expected flow, which wizards to base new work on, helper functions and a few UI guidelines. 

Furthermore, the page also includes a section around using telemetry in wizards.

I tried going from high-level to lower-level hints throughout the document. While we could be more specific or link to more examples in the second parts of the doc, I think we shouldn't specify too much, given that the wizard code can change at any time and we don't want to adjust develop too often. If reviewers would like to add more details, further helpful hints, feel free to suggest them. We can also add more specific hints to the wizard's `CONTRIBUTING.md` which lives much closer to the code. 

Side note: I think none of our wizards actually adhere to all of these specifications. This is fine though, as we can view them not just as a starting point for new wizards but also as a guide to improve existing wizards.

### CODE REVIEW

File: src/docs/sdk/setup-wizards/index.mdx [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+On a minimum, the wizard should print instructions to the console.
+
+If possible, the wizard should also add an example page or component to the project that throws an error.
+The implementation of such an example depends on the framework and is optional.
+
+## Building Blocks and Helper Functions
+
+When creating new wizards, it's recommended to base the wizard on [`clack`](https://github.com/natemoo-re/clack) which is a CLI library that provides nice abstractions around logging and input prompts.
+Many of the newer wizards, such as [NextJS](https://github.com/getsentry/sentry-wizard/tree/master/src/nextjs), are already using `clack`.
+It's recommended to these wizards as a reference.
```
Comment (brustolin) [inline src/docs/sdk/setup-wizards/index.mdx]:
```suggestion
It's recommended to use these wizards as a reference.
```

File: src/docs/sdk/setup-wizards/index.mdx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+Many of the challenges have already been solved.
+
+<Alert level="warning">
+  Our older wizards use a class and step-based approach which is not recommended anymore.
+  We want to keep things simple and avoid unnecessary abstractions going forward.
+</Alert>
+
+### Prompting for user input
+
+We already built a lot [helpers](https://github.com/getsentry/sentry-wizard/blob/master/src/utils/clack-utils.ts) around specific user input questions which we recommend to reuse and adapt as necessary.
```
Comment (brustolin) [inline src/docs/sdk/setup-wizards/index.mdx]:
```suggestion
We already built a lot of [helpers](https://github.com/getsentry/sentry-wizard/blob/master/src/utils/clack-utils.ts) around specific user input questions which we recommend to reuse and adapt as necessary.
```

**Review by brustolin — APPROVED** [review]
LGTM.
Just a couple of suggestions.

**Comment by smeubank** [comment]
great write up, it covers a lot and good level of detail. I don't have any feedback at present LGTM 👍

**Review by smeubank — APPROVED** [review]

**Review by romtsn — APPROVED** [review]
Looks great 💯

**Review by krystofwoldrich — APPROVED** [review]
Looks awesome!

The clean state check is handy, I believe RN Wizard doesn't do that at the moment.

**Pull request merged**